### PR TITLE
feat(evm_mlir): migrate `evm_mlir` project

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

Migrate https://github.com/lambdaclass/evm_mlir into `ethereum_rust`.

This is the first step, then we need to:
- https://github.com/lambdaclass/ethereum_rust/issues/550
- https://github.com/lambdaclass/ethereum_rust/issues/549.
- Migrate the PRs.

**Description**

Adds the https://github.com/lambdaclass/evm_mlir project as a crate for in `ethereum_rust`.

### CODE REVIEW

File: .github/workflows/evm_mlir.yaml [hunk 1]
```diff
@@ -0,0 +1,173 @@
+name: CI
```
Comment (mpaulucci) [inline .github/workflows/evm_mlir.yaml:1]: we need to only keep the evm_mlir specific stuff

File: crates/evm_mlir/.gitignore [hunk 2]
```diff
@@ -0,0 +1,12 @@
+**/target
```
Comment (mpaulucci) [inline crates/evm_mlir/.gitignore:1]: remove this

File: crates/evm_mlir/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+[workspace]
+members = ["bench/revm_comparison"]
+
+[package]
+name = "evm_mlir"
+version = "0.1.0"
+edition = "2021"
+repository = "https://github.com/lambdaclass/evm_mlir/"
```
Comment (mpaulucci) [inline crates/evm_mlir/Cargo.toml:8]: outdated

File: crates/evm_mlir/LICENSE [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (mpaulucci) [inline crates/evm_mlir/LICENSE:1]: remove

File: crates/evm_mlir/.gitignore (hunk 2 above)
Comment (jrchatruc) [reply crates/evm_mlir/.gitignore]: Removed the evm_mlir gitignore here [c800138](https://github.com/lambdaclass/ethereum_rust/pull/548/commits/c8001387fe182aaf26a52a1565ef772ae0a7f81c)

File: crates/evm_mlir/LICENSE (hunk 4 above)
Comment (jrchatruc) [reply crates/evm_mlir/LICENSE]: Removed in [958f57a](https://github.com/lambdaclass/ethereum_rust/pull/548/commits/958f57aa2de82a68751afadfe4c52d3897385073)

File: crates/evm_mlir/Cargo.toml (hunk 3 above)
Comment (jrchatruc) [reply crates/evm_mlir/Cargo.toml]: Updated file in [d3d0947](https://github.com/lambdaclass/ethereum_rust/pull/548/commits/d3d0947b30963a3a95463c67471cfb1cfe072a04)

File: .github/workflows/evm_mlir.yaml [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+          repo-name: llvm-repo
+          keys-asc: https://apt.llvm.org/llvm-snapshot.gpg.key
+      - name: Install LLVM
+        run: sudo apt-get install llvm-18 llvm-18-dev llvm-18-runtime clang-18 clang-tools-18 lld-18 libpolly-18-dev libmlir-18-dev mlir-18-tools
+
+      - name: Run cargo check
+        working-directory: ./crates/evm_mlir
+        run: cargo check --workspace --all-features --all-targets
+
+  lint:
```
[the remark this answers is not in the document]
Comment (jrchatruc) [reply .github/workflows/evm_mlir.yaml:56]: Same comment as above

File: .github/workflows/evm_mlir.yaml [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+on:
+  merge_group:
+  push:
+    paths:
+      - 'crates/evm_mlir/**'
+    branches: [ main ]
+  pull_request:
+    paths:
+      - 'crates/evm_mlir/**'
```
Comment (mpaulucci) [inline .github/workflows/evm_mlir.yaml:11]: nice! 🚀

File: crates/evm_mlir/README.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
+# EVM MLIR
+
+[![Telegram Chat][tg-badge]][tg-url]
+[![rust](https://github.com/lambdaclass/evm_mlir/actions/workflows/ci.yml/badge.svg)](https://github.com/lambdaclass/emv_mlir/actions/workflows/ci.yml)
```
Comment (mpaulucci) [inline crates/evm_mlir/README.md:4]: outdated

File: crates/evm_mlir/README.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+[![license](https://img.shields.io/github/license/lambdaclass/evm_mlir)](/LICENSE)
```
Comment (mpaulucci) [inline crates/evm_mlir/README.md:5]: same

File: crates/evm_mlir/README.md (hunk 7 above)
Comment (jrchatruc) [reply crates/evm_mlir/README.md]: Updated here [f72277f](https://github.com/lambdaclass/ethereum_rust/pull/548/commits/f72277fa1d720646b8788e7b1c16339c73d46889)

File: crates/evm_mlir/README.md (hunk 8 above)
Comment (jrchatruc) [reply crates/evm_mlir/README.md]: Updated here as well [f72277f](https://github.com/lambdaclass/ethereum_rust/pull/548/commits/f72277fa1d720646b8788e7b1c16339c73d46889)

**Review by mpaulucci — APPROVED** [review]

**Pull request merged**